xilem_web/src/concurrent/async_repeat.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{future::Future, marker::PhantomData, rc::Rc};
+
+use futures::channel::oneshot;
+use wasm_bindgen_futures::spawn_local;
+use xilem_core::{MessageResult, Mut, NoElement, View, ViewId, ViewMarker};
+
+use crate::{context::MessageThunk, DynMessage, Message, ViewCtx};
+
+pub fn async_repeat<M, F, H, State, Action, Fut>(
+    future_future: F,
+    on_event: H,
+) -> AsyncRepeat<F, H, M>
+where
+    F: Fn(AsyncRepeatProxy, oneshot::Receiver<()>) -> Fut + 'static,
+    Fut: Future<Output = ()> + 'static,
+    H: Fn(&mut State, M) -> Action + 'static,
+    M: Message,
+{
+    const {
+        assert!(
+            core::mem::size_of::<F>() == 0,
+            "`async_repeat` will not be ran again when its captured variables are updated.\n\
+            To ignore this warning, use `async_repeat_raw`."
+        );
+    };
+    AsyncRepeat {
+        future_future,
+        on_event,
+        message: PhantomData,
+    }
+}
+
+pub fn async_repeat_raw<M, F, H, State, Action, Fut>(
+    future_future: F,
+    on_event: H,
+) -> AsyncRepeat<F, H, M>
+where
+    F: Fn(MessageThunk) -> Fut + 'static,
+    Fut: Future<Output = ()> + 'static,
+    H: Fn(&mut State, M) -> Action + 'static,
+{
+    AsyncRepeat {
+        future_future,
+        on_event,
+        message: PhantomData,
+    }
+}
+
+pub struct AsyncRepeat<F, H, M> {
+    future_future: F,
```

Comment:
As Daniel said, I think
```suggestion
    init_future: F,
```

is probably more clear.